### PR TITLE
[ds-heap] binary heap implementation

### PR DESCRIPTION
**Pull request by pathfinder177**
* adding binary heap

### CODE REVIEW

File: heap.c [hunk 1]
```diff
@@ -0,0 +1,99 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdbool.h>
+#include "node.h"
```
Comment (Yanpas) [inline heap.c]: Do you need a node?

**Review by Yanpas — DISMISSED** [review]
I suggest you to review the API behind the heap. Most of the times we use heap to get "the largest (smallest) element: e.g. the timers queue. In this sense the heap behaves like a queue with push, pop and peek operations

File: heap.c (hunk 1 above)
Comment (pathfinder177) [reply heap.c]: Deleted

**Comment by pathfinder177** [comment]
> I suggest you to review the API behind the heap. Most of the times we use heap to get "the largest (smallest) element: e.g. the timers queue. In this sense the heap behaves like a queue with push, pop and peek operations

Began to implement

**Pull request merged**